### PR TITLE
feat: add uri scheme for opening timer by opening a link

### PR DESCRIPTION
**Pull request by olastor**
for #15

@tzugen can you please take a look? 

link scheme is like this: `bodhi://timer?times=60,120` (will start a 1 minute and a two minute timer)

Pausing does not work properly, but I think that is not necessary to support initially.

### CODE REVIEW

File: app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt [hunk 1]
```diff
@@ -7,17 +7,13 @@
 
 package org.yuttadhammo.BodhiTimer
 
-import android.content.ActivityNotFoundException
-import android.content.BroadcastReceiver
-import android.content.Context
-import android.content.Intent
-import android.content.IntentFilter
-import android.content.SharedPreferences
+import android.content.*
```
Comment (olastor) [inline app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt:10]: The IDE changed this, is the wildcard import a problem?

Comment (tzugen) [reply app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt:10]: It seems for some people this is important, I don't care either way

File: app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt [hunk 2]
```diff
@@ -108,6 +105,35 @@ class TimerActivity : AppCompatActivity(), View.OnClickListener, OnSharedPrefere
...
+
+            var notificationUri = prefs.getString("NotificationUri", Sounds.DEFAULT_SOUND)
+
+            when (notificationUri) {
+                "system" -> notificationUri = prefs.getString("SystemUri", "")
+                "file" -> notificationUri = prefs.getString("FileUri", "")
+            }
+
+            for (timeStr in data.getQueryParameter("times")!!.split(",").toTypedArray()) {
+                val time = Integer.valueOf(timeStr)
```
Comment (tzugen) [inline app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt:129]: Have you tested what happens
* when ``times=`` (empty param)
* when ``times=,``
* when ``times=@€(@,€(``

Comment (olastor) [reply app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt]: No, good point... The last two raise an unhandled error. I've added a try/catch and a check whether the timer list is empty after parsing.

**Comment by tzugen** [comment]
ah, forgot, could you add a short note to the readme to document this feature? Otherwise it will be very hidden.

**Comment by olastor** [comment]
> ah, forgot, could you add a short note to the readme to document this feature? Otherwise it will be very hidden.

sure, added in https://github.com/yuttadhammo/BodhiTimer/pull/80

**Pull request merged**